packages/components/src/number-control/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import BaseControl from '../base-control';
+
+function NumberControl( {
+	className,
+	label,
+	value,
+	instanceId,
+	id = `inspector-number-control-${ instanceId }`,
+	onChange,
+	help,
+	numberUnit,
+	...props
+} ) {
+	const onChangeValue = ( event ) => onChange( Number( event.target.value ) );
+
+	return (
+		<BaseControl
+			label={ label }
+			id={ id }
+			help={ help }
+			className={ classnames( 'components-number-control', className ) }
+		>
+			<input
+				className="components-number-control__input"
+				id={ id }
+				type="number"
+				onChange={ onChangeValue }
+				value={ value }
+				aria-label={ id ? undefined : label }
+				{ ...props }
+			/>
+
+			{ numberUnit && (
```

Comment:
Maybe we can update the Number control, to pass children to base control. It would allow us to remove the need for a numbeUnit prop making the component simpler. Current usages would only need to pass span prop with the unit like they do now.